src/fs/reader.ts

```diff
@@ -1,26 +1,32 @@
-import { IParserResult } from '@stoplight/types';
-import { parseWithPointers } from '@stoplight/yaml';
-import { existsSync, readFileSync } from 'fs';
-// @ts-ignore
-import * as fetch from 'node-fetch';
+import { readFile } from 'fs';
+const fetch = require('node-fetch');
+
+export const isURL = (uri: string) => /^https?:\/\//.test(uri);
 
 async function doRead(name: string, encoding: string) {
-  if (name.startsWith('http')) {
+  if (isURL(name)) {
```

Comment:
Todo: include `import` here to fall to require-ish flow and/or expose `registerRuleset` function to make integration look less hackier